### PR TITLE
Base tooltips refactor

### PR DESCRIPTION
**Pull request by samwinslow**
## Changes

Initial steps towards #4156. This PR introduces the basic components to allow for the custom tooltips as designed on #4156. One key consideration is that behavior is kept exactly the same for users without the feature flag (`4156-tooltips-legends`). Details:

- Introduces `InsightLabel` component to pretty print event/action names (e.g. Pageview instead of $pageview).
- Introduces the new tooltip component to be displayed when this feature flag is active. Example below,
    
- Introduces Chart JS crosshair plugin to show a single vertical line as shown on the mockups (might be discarded with the introduction of D3, but it's quite a simple change that there's no sense in undoing).

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,6 +52,7 @@
         "babel-preset-nano-react-app": "^0.1.0",
         "chart.js": "^2.9.3",
         "chartjs-adapter-dayjs": "^1.0.0",
+        "chartjs-plugin-crosshair": "^1.1.6",
```
Comment (paolodamico) [inline package.json:55]: Context on this @samwinslow ? We might no longer need it in this PR right? In general I'm thinking of getting this PR to a usable state and merge already as it's quite large as it is.

Comment (samwinslow) [reply package.json:55]: This renders the vertical scrubber line — without it, you need to hover each graph series individually and won't easily be able to see multiple series in the same tooltip.

Comment (paolodamico) [reply package.json:55]: But you're moving us to D3 anyways?

Comment (samwinslow) [reply package.json:55]: Yep! Can probably remove this for now and use whatever you'd like from the React components created in this branch.

**Comment by paolodamico** [comment]
@samwinslow there's one behavior I can't figure out and I could use your help, when you click in the graph body (as opposed clicking a specific point or the actual tooltip) it treats it as a click in the users point, do you have some context on how to fix this? FYI @mariusandra this is what's causing the cypress component test failure, we'll fix before merging, but a review is already helpful.

Failing E2E tests are flaky and pass locally.

**Review by mariusandra — CHANGES_REQUESTED** [review]
This looks very nice, though here's what I found:

- It's possible to click to the closest point, not the open tooltip:


- Throws some noise for bar(value) charts and shows no label


- The colors seem to have gone missing for bar(time) charts
image

- Switching from bar chart to line chart crashes everything


- Opening a pie chart crashes everything

**Comment by mariusandra** [comment]
@paolodamico that graph body click issue happens because there's always one point selected on the graph.

**Comment by paolodamico** [comment]
Thanks for the quick yet thorough review @mariusandra! Everything has been addressed, except point 1 (clicking only on the tooltip body). The reason for this is that this requires a somewhat complex workaround that is purely `chart-js` based. Given that we're moving to D3 I don't think it's worth spending time on this right now, but feel free to push back.

**Review by mariusandra — APPROVED** [review]
Seems like the workaround was quite easy :). LGTM!

**Comment by paolodamico** [comment]
Thanks @mariusandra! I see, I actually misunderstood you then. What I was trying to accomplish was that if you click on an empty space in the graph, nothing happens. We only open the users modal when you click on either the exact point or the tooltip container, but we can definitely address this later.

**Pull request merged**